examples/2d/multiple_sprites_one_entity.rs

```diff
@@ -0,0 +1,92 @@
+//! Constructs a large entity composed of multiple sprites
+
+use bevy::input::common_conditions::input_pressed;
+use bevy::prelude::*;
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins.set(ImagePlugin::default_nearest())) // prevents blurry sprites
+        .add_systems(Startup, setup)
+        // the user can press the arrow keys to translate the parent entity in the window
+        .add_systems(
+            Update,
+            translate::<-5, 0>.run_if(input_pressed(KeyCode::ArrowLeft)),
+        )
+        .add_systems(
+            Update,
+            translate::<5, 0>.run_if(input_pressed(KeyCode::ArrowRight)),
+        )
+        .add_systems(
+            Update,
+            translate::<0, 5>.run_if(input_pressed(KeyCode::ArrowUp)),
+        )
+        .add_systems(
+            Update,
+            translate::<0, -5>.run_if(input_pressed(KeyCode::ArrowDown)),
+        )
```

Comment:
I would very much prefer if this used a single system and input handling being done in that system.

If you really want to do it this way, you should use a single `.add_systems`

```suggestion
        .add_systems(
            Update,
            (
                translate::<-5, 0>.run_if(input_pressed(KeyCode::ArrowLeft)),
                translate::<5, 0>.run_if(input_pressed(KeyCode::ArrowRight)),
                translate::<0, 5>.run_if(input_pressed(KeyCode::ArrowUp)),
                translate::<0, -5>.run_if(input_pressed(KeyCode::ArrowDown)),
            ),
        )
```

Reply:
Why would you prefer input handling inside the system itself? Just for readability?

There are other `examples` already which use const generics. `virtual_time`, for example